Jenkins_setup.md

```diff
@@ -0,0 +1,161 @@
+# Kata Containers Jenkins CI setup
+
+The main Continuous Integration (CI) system for Kata Containers is [Jenkins](https://jenkins.io/).
+This document gives an overview of how Kata Containers integrates with Jenkins, and
+provides pointers and information to configuration information and scripts that will
+help set up a Kata Containers Jenkins CI system.
+
+## Types of jobs
+
+Kata Containers CI operates two types of CI builds:
+
+- PR builds. Builds are scheduled upon submission or change to a Pull Request on any
+of the repositories
+
+- Master builds. A build is scheduled upon the merge of a Pull Request into the master
+branch of a repository.
+
+How these builds are configured and executed are subtly different:
+
+### PR builds
+
+PR builds are triggered from Jenkins using the [GitHub Pull Request Builder](https://plugins.jenkins.io/ghprb) (GHPRB)
+plugin.
+
+The GHPRB plugin is configured to be triggered via github hooks. This is the preferred method
+to trigger on changes (preferred to polling). The preferred method is to integrate the CIs
+using the [katacontainersbot](https://github.com/katacontainersbot) user. Please discuss
+your requirements with the Kata Containers team who can then discuss the appropriate changes.
+
+The GHPRB triggers should be set up similar to:
+![Jenkins GHPRB plugin trigger config](pictures/ghprb_trigger.png)
+
+The GHPRB `Trigger Setup` further options should be set up similar to:
+![Jenkins GHPRB plugin trigger setup config](pictures/ghprb_adv_triggers.png)
+
+### Master builds
+
+Master branch builds are tracked with the Jenkins [git Source Code Management](https://plugins.jenkins.io/git)
+plugin. GITScm GitHub hooks are used to trigger builds.
+
+The Jenkins setup should look similar to the following for the:
+
+- Repository URL setup
+- Branches to build
+- Hook trigger enablement
+
+![Jenkins master setup](pictures/master_git_config.png)
+
+
+### Github bandwidth and tokens
+
+It should be noted that the GitHub API access is [rate limited](https://developer.github.com/v3/#rate-limiting). If no github access authentication method is configured into Jenkins
+then the CI may quickly run out of GitHub access bandwidth, which will stall the CI.
+
+For Kata Containers CI we configure Jenkins with a GitHub [Personal API token](https://blog.github.com/2013-05-16-personal-api-tokens/), which increases the bandwidth capacity.
+
+## Job build script
+
+Kata Containers Jenkins builds, for both PR and Master builds, are invoked using the top
+level [`jenkins_job_build.sh`](https://github.com/kata-containers/tests/blob/master/.ci/jenkins_job_build.sh) script from the [katacontainers/tests](https://github.com/kata-containers/tests)
+repository.
+
+This script takes a single argument (the path to the repository to build), and utilises
```

Comment:
Lines 63-64 suggested rewrite: This script takes a single argument (i.e. the path to the repository to build), and utilizes some environment variables to guide what sort of build and run it will perform.